rmgpy/data/thermo.py

```diff
@@ -1687,11 +1606,98 @@ def lowest_energy(species):
+        and rotational degrees of freedom from the gas phase.
+
+        Args:
+            adsorption_thermo ([ThermoData]): the ThermoData object to add the correction(s)
+            adsorption_groups ([database]): the groups database (adsorptionPt111)
+            molecule ([Molecule]): the molecule to apply the thermo correction
+            surface_sites ([list([Atom])]): a list of the surface site atoms in the molecule
+        """
+
+        number_of_surface_sites = len(surface_sites)
+
+        matches = []
+        for atom in surface_sites:
+            labeled_atoms = {'*': atom}
+            node0 = adsorption_groups.descend_tree(molecule, labeled_atoms)
+            if node0 is None: 
+                # no match, so try the next surface site
+                continue
+            node = node0
+            while node is not None and node.data is None:
+                node = node.parent
+            if node is None:
+                # no data, so try the next surface site
+                continue
+            data = node.data
+            comment = node.label
+            loop_count = 0
+            while isinstance(data, str):
+                loop_count += 1
+                if loop_count > 100:
+                    raise DatabaseError("Maximum iterations reached while following thermo group data pointers. A circular"
+                                    f" reference may exist. Last node was {node.label} pointing to group called {data} in "
+                                    f"database {database.label}")
+
+                for entry in adsorption_groups.entries.values():
+                    if entry.label == data:
+                        data = entry.data
+                        comment = entry.label
+                        break
+                else:
+                    raise DatabaseError(f"Node {node.label} points to a non-existing group called {data} "
+                                    f"in database {database.label}")
+            data.comment = f'{adsorption_groups.label}({comment})'
+            group_surface_sites = node.item.get_surface_sites()
+            if len(group_surface_sites) == number_of_surface_sites:
+                # all the surface sites are accounted for so add the adsorption group and return
+                add_thermo_data(adsorption_thermo, data, group_additivity=True)
+                return True
+            else:
+                # we have not found a full match yet, so append and keep looking
+                matches.append((len(group_surface_sites),data))
+        
+        matches.sort(key = lambda x: -x[0])
+        # sort the matches by descending number of surface sites
+        for i,(number_of_group_sites, data) in enumerate(matches):
+            if number_of_surface_sites - number_of_group_sites < 0:
+                # too many sites in this group, skip to the next one
+                continue
+            if i == 0:
```

Comment:
Im not sure if this could happen with the current database, but maybe it's possible.  I added a `corrections_applied` counter here https://github.com/ReactionMechanismGenerator/RMG-Py/pull/2169/commits/0455cfaa0a93abfe1316d4fb8535ca2aa6de09d1 just in case